ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.app
+
+import com.daml.testing.postgresql.{PostgresAround, PostgresAroundAll}
+import org.mockito.ArgumentMatchers._
+import org.mockito.MockitoSugar._
+import org.scalatest.{AsyncFlatSpec, Matchers}
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+final class RunnerSpec
+    extends AsyncFlatSpec
+    with Matchers
+    with PostgresAround
+    with PostgresAroundAll {
+
+  private val MockedLedgerFactory = mock[LedgerFactory[ReadWriteService, Unit]]
+  when(MockedLedgerFactory.defaultExtraConfig).thenReturn(())
+  when(MockedLedgerFactory.manipulateConfig(any[Config[Unit]])).thenAnswer[Config[Unit]](identity)
+
+  private val DumpIndexMetadataCommand = "dump-index-metadata"
+
+  behavior of "Runner"
+
+  it should "fail if a participant is not provided in run mode" in {
+
+    val runner = new Runner("Test", MockedLedgerFactory).owner(Seq.empty)
+    runner.use(_ => Future.unit).map(_ => fail).recover { case NonFatal(_) => succeed }
```

Comment:
Perhaps use `.failed` here and below?

```suggestion
    runner.use(_ => Future.unit).failed.map(_ => succeed)
```

Reply:
Addressed by https://github.com/digital-asset/daml/pull/7213/commits/6e8283ff951a96f6b60752a9b86dbc47ec948e39